### PR TITLE
Benchmarks dev guide

### PR DESCRIPTION
**Pull request by MiguelWeezardo**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-benchto-benchmarks/.java-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.8
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/.java-version:1]: What uses this file? Shouldn't it be enough to set the required JDK in the POM?

File: testing/trino-benchto-benchmarks/README.md [hunk 2]
```diff
@@ -48,8 +52,10 @@ data-sources:
 environment:
   name: TRINO-DEV
 
-presto:
+trino:
   url: http://localhost:8080
+  username: na
+  password: na
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/README.md:58]: The password should not be required by default. Maybe change the example user to `benchto` so it's a bit more informative than `na`.

File: testing/trino-benchto-benchmarks/README.md [hunk 3]
```diff
@@ -84,10 +90,20 @@ tpch_medium: tpcds_10gb_txt
...
 With the scene set up as in the previous section, the benchmark can be run with:
 ```bash
-./mvnw clean package -pl :trino-benchto-benchmarks
-java -Xmx1g -jar trino-benchto-benchmarks/target/trino-benchto-benchmarks-*-executable.jar \
-    --sql trino-benchto-benchmarks/src/main/resources/sql \
-    --benchmarks trino-benchto-benchmarks/src/main/resources/benchmarks \
-    --activeBenchmarks=presto/tpch --profile=presto-devenv \
-    --overrides overrides.yaml
+# set JAVA_HOME to the JDK 8 home dir
+local_repo=$(./mvnw -B help:evaluate -Dexpression=settings.localRepository -q -DforceStdout)
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/README.md:94]: Can we assume it's always `$HOME/.m2/repository`? That'll make the example simpler.

File: testing/trino-benchto-benchmarks/README.md [hunk 4]
```diff
@@ -84,10 +90,20 @@ tpch_medium: tpcds_10gb_txt
...
+java -Xmx1g \
+            -jar "$local_repo/io/trino/benchto/benchto-driver/0.18/benchto-driver-0.18.jar" \
+            --profiles.directory "." \
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/README.md:97]: This should not be necessary, the `application.yaml` needs to be in the current directory anyway. Maybe also remove other flags, like `-Xmx1g` and the time limit and paths to sql and benchmark dirs.

File: testing/trino-benchto-benchmarks/README.md [hunk 5]
```diff
@@ -84,10 +90,20 @@ tpch_medium: tpcds_10gb_txt
...
+            --benchmarks src/main/resources/benchmarks \
+            --activeBenchmarks=presto/tpch \
+            --overrides "overrides.yaml" \
+            --timeLimit PT20M
 ```
+
+
+### Automated benchmarks
+
+A complete script which runs a small suite of benchmarks can be found in `perftest/benchto.sh`.
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/README.md:108]: This doesn't exist (yet).

File: testing/trino-benchto-benchmarks/README.md [hunk 6]
```diff
@@ -48,8 +52,10 @@ data-sources:
 environment:
   name: TRINO-DEV
 
-presto:
+trino:
```
Comment (nineinchnick) [inline testing/trino-benchto-benchmarks/README.md:55]: You have to also either change the data source name to `presto` or update all benchmarks.

File: testing/trino-benchto-benchmarks/.java-version (hunk 1 above)
Comment (MiguelWeezardo) [reply testing/trino-benchto-benchmarks/.java-version:1]: POM only impacts Maven, but in the README we tell developers to run java, so we can leverage jenv if they have it.

File: testing/trino-benchto-benchmarks/README.md (hunk 5 above)
Comment (MiguelWeezardo) [reply testing/trino-benchto-benchmarks/README.md:108]: Isn't the script developed on the `benchmarks` branch where this PR is aimed at?

File: testing/trino-benchto-benchmarks/README.md (hunk 6 above)
Comment (MiguelWeezardo) [reply testing/trino-benchto-benchmarks/README.md:55]: Guess I'll have to return to the old name. We can do the rename in a separate commit/PR

File: testing/trino-benchto-benchmarks/README.md (hunk 5 above)
Comment (nineinchnick) [reply testing/trino-benchto-benchmarks/README.md:100]: Yes, but it should be separated from that effort so we can get this in faster.

File: testing/trino-benchto-benchmarks/.java-version (hunk 1 above)
Comment (nineinchnick) [reply testing/trino-benchto-benchmarks/.java-version:1]: The README should say explicitly which JDK version to use, without assuming how users manage their JDK installations.

**Comment by MiguelWeezardo** [comment]
Making a separate PR not based on `benchmarks` branch

**Comment by MiguelWeezardo** [comment]
Continued on https://github.com/trinodb/trino/pull/13988

**Pull request closed without merging**